Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/UserDataActivity.kt

```diff
@@ -93,11 +96,22 @@ class UserDataActivity : AppCompatActivity() {
             dialog.arguments = arguments
             dialog.show(supportFragmentManager, UserDataImportWarningDialog.TAG)
         } else if (requestCode == REQUEST_CODE_EXPORT && resultCode == RESULT_OK) {
-            ThreadUtil.runOnThreadAndShowResult(
-                this,
-                R.string.export_in_progress,
-                0,
-                { resources.getString(exportUserData(data!!.data!!)) })
+            taskViewModel = ViewModelProvider(this)[TaskViewModel::class.java]
+            taskViewModel.clear()
+            taskViewModel.resultGetter = {
+                val resultResource = exportUserData(data!!.data!!)
+                taskViewModel.result.postValue(resultResource)
+                taskViewModel.isComplete.postValue(true)
```

Comment:
Is there a reason why `TaskViewModel` doesn't handle setting `isComplete` on its own?